drivers/soundwire/stream.c

```diff
@@ -1544,23 +1549,45 @@ static int _sdw_prepare_stream(struct sdw_stream_runtime *stream)
  */
 int sdw_prepare_stream(struct sdw_stream_runtime *stream)
 {
+	bool update_params = true;
 	int ret = 0;
 
+	pr_err("%s: %s start\n", __func__, stream->name);
+
 	if (!stream) {
 		pr_err("SoundWire: Handle not found for stream\n");
 		return -EINVAL;
 	}
 
-	if (stream->state != SDW_STREAM_CONFIGURED)
-		return 0;
-
 	sdw_acquire_bus_lock(stream);
 
-	ret = _sdw_prepare_stream(stream);
+	if (stream->state != SDW_STREAM_CONFIGURED &&
+	    stream->state != SDW_STREAM_DISABLED) {
```

Comment:
@plbossart Will it be better if we do the check before sdw_acquire_bus_lock() and move state_err: after sdw_release_bus_lock()?

Reply:
@bardliao I think we have to use the lock before checking the state.
Put differently, the state is changed when the bus is locked, so it would be odd to allow the state to be checked without taking the lock as well, you could have race conditions when you read a state that becomes immediately inconsistent.